src/annotator/anchoring/range-js.js

```diff
@@ -32,3 +32,42 @@ export function nodeFromXPath(xpath, root) {
   });
   return node;
 }
+
+/**
+ * Finds an element node using an XPath relative to the document root.
+ *
+ * Example:
+ *   node = nodeFromXPath('/html/body/div/p[2]')
+ */
+export function nodeFromXPath(xpath, root = document) {
+  /**
+   * Attempt to evaluate a provided XPath. If the evaluation fails, then fall back to a
+   * manual evaluation using `nodeFromXPathFallback` that can evaluate very simple XPaths such
+   * as those generated by `xpathFromNode`
+   */
+  function evaluateXPath(xp, root, nsResolver = null) {
+    try {
+      return document.evaluate(
+        '.' + xp,
+        root,
+        nsResolver,
+        XPathResult.FIRST_ORDERED_NODE_TYPE,
+        null
+      ).singleNodeValue;
+    } catch (e) {
+      // There are cases when the evaluation fails, because the
```

Comment:
As per our voice chat, can you rewrite this long comment to make a little more sense and add a note that we haven't yet found a test case that will cause this exception?